### PR TITLE
Enhance

### PR DESCRIPTION
**Pull request by armanoptimum**

### CODE REVIEW

File: src/components/Banner/styles.js [hunk 1]
```diff
@@ -82,27 +86,54 @@ export const BannerImageContainerStyled = styled.div`
...
+  background-color: var(--color-primary-dark-blue);
+  transform: rotate(90deg);
+  border-radius: 3rem;
+  z-index: 2;
+
+  @media (min-width: 60rem) {
+    width: 30rem;
+    height: 30rem;
+  }
+`;
```
Comment (NoorAbuzainahOP) [inline src/components/Banner/styles.js:139]: these two components roughly look the same, why not reuse a BaseBanner and override the bg color?

File: src/pages/product/[id]/_data.js
[the remark this answers is not in the document]
Comment (armanoptimum) [reply src/pages/product/[id]/_data.js:1]: @NoorAbuzainahOP  fetch function move to index.js?

[the remark this answers is not in the document]
Comment (NoorAbuzainahOP) [reply src/pages/product/[id]/_data.js:1]: yes, the whole file path

File: src/pages/_data/index.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  return { data, total };
+};
+
+export const fetchProductData = async (id) => {
+  try {
+    const res = await fetch(`${baseUrl}/api/product/${id}`);
+    if (!res.ok) {
+      throw new Error('Product not found');
+    }
+    return await res.json();
```
Comment (NoorAbuzainahOP) [inline src/pages/_data/index.js:15]: does `json()` here return a promise? do we need the `await`?

File: src/pages/_data/index.js [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  } catch (e) {
+    console.error(e);
+    return null;
+  }
+};
+
+export const fetchCarousel = async () => {
+  try {
+    const res = await fetch(`${baseUrl}/api/carousel`);
+    return await res.json();
```
Comment (NoorAbuzainahOP) [inline src/pages/_data/index.js:25]: same here

File: src/components/Filter/NumberFilter/index.jsx [hunk 4]
```diff
@@ -20,28 +21,27 @@ const NumberFilter = ({ children }) => {
...
-    }
-
-    if (range.max) {
-      currentQuery.set('max', range.max);
-    } else {
-      currentQuery.delete('max');
-    }
-
+    updateQueryParamNumber(currentQuery, 'min', range.min);
+    updateQueryParamNumber(currentQuery, 'max', range.max);
```
Comment (NoorAbuzainahOP) [inline src/components/Filter/NumberFilter/index.jsx:25]: same as the comment in src/components/Filter/CheckboxFilter/index.jsx , please pass the filter name dynamically

File: src/pages/product/layout.js
Comment (NoorAbuzainahOP) [inline src/pages/product/layout.js:1]: please move this to `src/layouts/product` directory

File: src/pages/_data/index.js (hunk 2 above)
Comment (armanoptimum) [reply src/pages/_data/index.js:15]: json method returns promise that's why i used await.

Comment (NoorAbuzainahOP) [reply src/pages/_data/index.js:15]: @armanoptimum if json() returns a promise and we are using await inside the function, what is the need of the other await when calling this function [here](https://github.com/armanoptimum/pages-pet-fe/blob/30a0555135c117fc4c717320f8ab3f2bbe511ca0/src/pages/product/%5Bid%5D/index.jsx#L10)?

File: src/components/Cards/CardSecondary/styles.js
Comment (NoorAbuzainahOP) [inline src/components/Cards/CardSecondary/styles.js:1]: some of the styled components in this file looks almost identical to the styled components in `src/components/Cards/CardPrimary/styles.js`, please move any identical components to `src/components/Cards/CardSecondary/styles.js` and reuse them

File: src/components/Carousel/index.js
Comment (NoorAbuzainahOP) [inline src/components/Carousel/index.js:1]: did you mean to name the file `index.jsx`?

File: src/pages/_data/index.js (hunk 2 above)
Comment (armanoptimum) [reply src/pages/_data/index.js:15]: yes only make sense here  const { data, total } = await res.json();

**Pull request merged**